source/rock/middle/TypeDecl.ooc

```diff
@@ -1253,6 +1253,29 @@ TypeDecl: abstract class extends Declaration {
                     varAcc := VariableAccess new("this", access token)
                     varAcc reverseExpr = access
                     access expr = varAcc
+                    if (res params explicitThis) {
+                        mustBeExplicit := true
+
+                        // Generic variables do not require explicit this
+                        if (vDecl getType()) {
+                            if (vDecl getType() toString() == "Class") {
+                                mustBeExplicit = false
+                            }
+                        }
+
+                        // Access to a variable from within its own declaration does not have to be explicit
+                        depth := trail getSize() - 1
+                        while (depth >= 0) {
+                            if (trail get(depth) == vDecl) {
+                                mustBeExplicit = false
```

Comment:
Filter out access from inside the variable's own declaration since most of them are generated getters and setters.